aggregator/handler/common/queue.go

```diff
@@ -96,163 +104,163 @@ type forwardHandler struct {
-	h.RLock()
-	if h.closed {
-		h.RUnlock()
+func (q *queue) enqueue(buffer *RefCountedBuffer, dropType dropType) error {
+	q.RLock()
+	if q.closed {
+		q.RUnlock()
 		buffer.DecRef()
-		h.metrics.enqueueErrors.Inc(1)
-		return errHandlerClosed
+		q.metrics.enqueueErrors.Inc(1)
+		return errQueueClosed
 	}
 	for {
 		select {
-		case h.bufCh <- buffer:
-			h.RUnlock()
-			h.metrics.enqueueSuccesses.Inc(1)
+		case q.bufCh <- buffer:
+			q.RUnlock()
+			q.metrics.enqueueSuccesses.Inc(1)
 			return nil
 		default:
 			if dropType == dropCurrent {
-				h.RUnlock()
+				q.RUnlock()
 				buffer.DecRef()
-				h.metrics.enqueueErrors.Inc(1)
-				h.metrics.currentDropped.Inc(1)
-				return errBufferQueueFull
+				q.metrics.enqueueErrors.Inc(1)
+				q.metrics.currentDropped.Inc(1)
+				return errQueueFull
 			}
 		}
 
 		// Drop oldest in queue to make room for new buffer.
 		select {
-		case buf := <-h.bufCh:
+		case buf := <-q.bufCh:
 			buf.DecRef()
-			h.metrics.oldestdDropped.Inc(1)
+			q.metrics.oldestdDropped.Inc(1)
 		default:
 		}
 	}
 }
 
-func (h *forwardHandler) forwardToConn(addr string, conn *net.TCPConn) {
-	defer h.wg.Done()
+func (q *queue) forwardToConn(addr string, conn *net.TCPConn) {
+	defer q.wg.Done()
 
 	var err error
 	continueFn := func(int) bool {
-		h.RLock()
-		closed := h.closed
-		h.RUnlock()
+		q.RLock()
+		closed := q.closed
+		q.RUnlock()
 		return !closed
 	}
 	connectFn := func() error {
-		conn, err = h.tryConnectFn(addr)
+		conn, err = q.tryConnectFn(addr)
 		return err
 	}
 
 	for {
-		// Retry establishing connection until either success or the handler is closed.
+		// Retry establishing connection until either success or the queue is closed.
 		for conn == nil {
-			attemptErr := h.reconnectRetrier.AttemptWhile(continueFn, connectFn)
+			attemptErr := q.reconnectRetrier.AttemptWhile(continueFn, connectFn)
 			if attemptErr == retry.ErrWhileConditionFalse {
 				return
 			}
 		}
 
 		// Dequeue buffers and forward to server.
 		for {
-			buf, ok := <-h.bufCh
+			buf, ok := <-q.bufCh
 			if !ok {
 				conn.Close()
-				atomic.AddInt32(&h.numOpenConnections, -1)
+				atomic.AddInt32(&q.numOpenConnections, -1)
 				return
 			}
-			_, writeErr := h.writeToConnFn(conn, buf.Buffer().Bytes())
+			_, writeErr := q.writeToConnFn(conn, buf.Buffer().Bytes())
 			if writeErr == nil {
-				h.metrics.writeSuccesses.Inc(1)
+				q.metrics.writeSuccesses.Inc(1)
 				buf.DecRef()
 				continue
 			}
-			h.metrics.writeErrors.Inc(1)
-			h.log.WithFields(
+			q.metrics.writeErrors.Inc(1)
+			q.log.WithFields(
 				log.NewField("address", addr),
 				log.NewErrField(writeErr),
 			).Error("error writing to server")
 
 			// NB(xichen): the buffer contains the oldest flushed data in queue
 			// so it's preferrable to drop it in case the queue is full.
-			if enqueueErr := h.enqueue(buf, dropCurrent); enqueueErr != nil {
-				h.log.WithFields(
+			if enqueueErr := q.enqueue(buf, dropCurrent); enqueueErr != nil {
```

Comment:
nit: same as above, maybe just `err`?

Reply:
Yes the metrics have already been timestamped at this point. Sounds good will rename to `err`.